src/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -842,6 +842,23 @@ public static TSource SingleOrDefault<TSource>(this IQueryable<TSource> source,
                     ));
         }
 
+        public static bool TrySingle<TSource>(this IQueryable<TSource> source, out TSource element)
+        {
+            Tuple<bool, TSource> result = TrySingle(source);
+            element = result.Item2;
+            return result.Item1;
+        }
+
+        internal static Tuple<bool, TSource> TrySingle<TSource>(this IQueryable<TSource> source)
+        {
+            if (source == null)
+                throw Error.ArgumentNull(nameof(source));
+            return source.Provider.Execute<Tuple<bool, TSource>>(
+                Expression.Call(
+                    null,
+                    CachedReflectionInfo.TrySingle_TSource_1(typeof(TSource)), source.Expression));
+        }
+
```

Comment:
`TrySingle` should probably work analogously to the other query methods; produce an expression representing a call of themselves on the expression of the source queryable. That way providers work on knowing what "TrySingle" means (or don't and fail, but the failure is on them). In particular the `EnumerableQuery` class will rewrite the call to `Queryable.TrySingle` as a call to `Enumerable.TrySingle` on the wrapped enumerable, and then work correctly.